software/contrib/consequencer.py

```diff
@@ -0,0 +1,310 @@
+        if self.analogInputMode != 2:
+            self.pattern = k2.read_position(len(self.BD))
+        else:
+            # Get the analogue input voltage as a percentage
+            val = 100 * ain.percent()
+        
+            # Is there a voltage on the analogue input and are we configured to use it?
+            if val > 0.4:
+                # Convert percentage value to a representative index of the pattern array
+                self.pattern = int((len(self.BD) / 100) * val)
+            else:
+                self.pattern = k2.read_position(len(self.BD))
+        
+        #self.step_length = 16
+        self.step_length = len(self.BD[self.pattern])
+        #print('Pattern: ' + str(self.pattern) + ' Step Length: ' + str(self.step_length))
+
+    def getCvPattern(self):
+
+        # If analogue input mode 3, get the CV pattern from CV input
+
+        if self.analogInputMode != 3:
+            return
+        else:
+            # Get the analogue input voltage as a percentage
+            CvpVal = 100 * ain.percent()
+            
+            # Is there a voltage on the analogue input and are we configured to use it?
+            if CvpVal > 0.4:
+                # Convert percentage value to a representative index of the pattern array
+                self.CvPattern = int((len(self.random4) / 100) * CvpVal)
+
+    def generateRandomPattern(self, length, min, max):
+        self.t=[]
+        for i in range(0, length):
+            self.t.append(uniform(0,9))
+        return self.t
+
+
+    def getRandomness(self):
+        # If not mode 1, get the value from the knob position
+        if self.analogInputMode != 1:
+            self.randomness = k1.read_position()
+        else:
+            # Check if there is CV on the Analogue input, if not use the k1 position
+            val = 100 * ain.percent()
+            if val > 0.4:
+                self.randomness = val
+            else:
+                self.randomness = k1.read_position()
+
+        #print(rCvVal)
+        #print(ain.read_voltage())
+
+    def main(self):
+        while True:
+
+            #self.setClockDivision()
+            self.getPattern()
+            self.getRandomness()
+            self.getCvPattern()
+            self.updateScreen()
+            self.reset_timeout = 500
+            # If I have been running, then stopped for longer than reset_timeout, reset the steps and clock_step to 0
+            #if self.clock_step != 0 and ticks_diff(ticks_ms(), self.last_clock_input) > self.reset_timeout:
+            #print('din.last_triggered: ' + str(din.last_triggered()))
+            #print('Step: ' + str(self.step))
+            if self.clock_step != 0 and ticks_diff(ticks_ms(), din.last_triggered()) > self.reset_timeout:
+                #print('Resetting...')
+                self.step = 0
+                self.clock_step = 0
+            #sleep_ms(100)
+
+    def setClockDivision(self):
+        k1Val = k1.read_position()
```

Comment:
Consider simplifying this method by leveraging [Knob.choice()](https://allen-synthesis.github.io/EuroPi/generated/europi.html#europi.Knob.choice) or remove it (it looks like its references are commented out).
```suggestion
    return k1.choice([1, 2, 3, 4, 5, 6, 7, 8, 16, 32])
```